src/tox/_pytestplugin.py

```diff
@@ -49,6 +49,15 @@ def work_in_clean_dir(tmpdir):
         yield
 
 
+@pytest.fixture(autouse=True)
+def check_cwd_not_changed_by_test():
+    old = os.getcwd()
+    yield
+    new = os.getcwd()
+    if old != new:
+        pytest.fail("test changed cwd: {!r} => {!r}".format(old, new))
+
+
```

Comment:
@gaborbernat 
Since this adds overhead to every test I've not meant to keep it - maybe comment it out / only use it on CI?